simpleble/src_c/peripheral.cpp

```diff
@@ -15,6 +15,18 @@ void simpleble_peripheral_release_handle(simpleble_peripheral_t handle) {
     delete peripheral;
 }
 
+char* simpleble_peripheral_address_type(simpleble_peripheral_t handle) {
+    if (handle == nullptr) {
+        return nullptr;
+    }
+
+    SimpleBLE::Safe::Peripheral* peripheral = (SimpleBLE::Safe::Peripheral*)handle;
+    std::string address_type = peripheral->address_type().value_or("");
+    char* c_address_type = (char*)malloc(address_type.size() + 1);
```

Comment:
Created https://github.com/OpenBluetoothToolbox/SimpleBLE/issues/136 to track future simplifications of this logic.